### PR TITLE
[docs] Introduce core concepts for CDC 3.0 pipeline

### LINKED ISSUE #2655: [docs] Introduce core concepts for CDC 3.0 pipeline
### Search before asking

- [X] I searched in the [issues](https://github.com/ververica/flink-cdc-connectors/issues) and found nothing similar.

### Motivation

Introduce core concepts for CDC 3.0 pipeline

### Solution

_No response_

### Alternatives

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by gtk96**
closed #2655

### CODE REVIEW

**Comment by qidian99** [comment]
In #2719, there is Chinese documentation counterpart. I suppose the Chinese version of core concepts has already been merged? 

o/w LGTM. cc @leonardBang

File: docs/content/cdc-3.0-core-concepts.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Routing, for each change event, specifies the target table ID of the event. The most typical application scenario is the merge of sub-databases and sub-tables, routing multiple upstream source tables to the same result table.
+
+- source-table: source table id, supports regular expressions
+- sink-table: Result table id, supports regular expressions
+- description: Routing rule description, user-defined (optional, default value provided)
+
+```yml
+route:
+  - source-table: mydb.app_order_.*
```
Comment (leonardBang) [inline docs/content/cdc-3.0-core-concepts.md:71]: should be `mydb.app_order_\.*`, if the table regex pattern contains `.`, we need to use `\.` to escape

File: docs/content/cdc-3.0-core-concepts.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+Data transformation specifies how to transform the change data for each row in a given table:
+
+- Source table: The ID of the source table, supports regular expressions.
+- SQL expression: Evaluates an expression on the data in the source table.
+- Description: A description of the transformation expression, which is user-defined (optional, with a default value provided).
+
+```yml
+transform:
+  - source-table: mydb.app_order_.*
```
Comment (leonardBang) [inline docs/content/cdc-3.0-core-concepts.md:51]: ditto

File: docs/content/cdc-3.0-core-concepts.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    projection: id, order_id, TO_UPPER(product_name)
+    filter: id > 10 AND order_id > 100
+    description: project fields from source table    	# Optional parameter for description purpose
+  - source-table: mydb.web_order_.*
```
Comment (leonardBang) [inline docs/content/cdc-3.0-core-concepts.md:55]: ditto

**Comment by loserwang1024** [comment]
>  I suppose the Chinese version of core concepts has already been merged?
I cannot find it.

**Comment by gtk96** [comment]
Corresponding issue has been closed, close this pr

**Pull request closed without merging**